macos/FluentUIUnitTest/AvatarViewTests.swift

```diff
@@ -115,13 +115,16 @@ class AvatarViewTests: XCTestCase {
 		XCTAssertNil(noNameNoEmailAvatar.toolTip)
 	}
 
-	func testBackgroundColors () {
+	func testAvatarColors () {
```

Comment:
Is there a way to test dynamic colors?
I think it's very valuable to have the colorsets compared in dark and light mode since the background and foreground values are just swapped when going from light to dark mode.

Reply:
Unfortunately, there isn't a trivial way to test without using a dependency injection framework like [OcMock](https://ocmock.org/swift/). We need a way to override theme values which isn't currently supported API. 

If you have any ideas, I can look into it.

Reply:
I think based on the issue I just reported on the other comment about light colors being displayed in both light and dark modes, we can follow an approach like:

 1. create a type that holds the 2 values of the color and returns the correct color in a computed property ([resolvedColor](https://developer.apple.com/documentation/uikit/uicolor/3238042-resolvedcolor) to mimic the iOS API)
 
 2. Subscribe to the effective appearance change inside the Avatar code so you can update the visuals of it when it changes from aqua to darkAqua (calling updateAvatarViewContents() whenever it changes)
 
 Something like this example:
 https://developer.apple.com/forums/thread/104515
 ```
private var appearanceObserver: NSKeyValueObservation?

  override func viewWillMove(toWindow newWindow: NSWindow?) {
       super.viewWillMove(toWindow: newWindow)
       if let window = newWindow {
            appearanceObserver = window.observe(\.effectiveAppearance) { [weak self] (window, change) in
                 self?.updateAppearance(forWindow: window)
            }
       } else {
            appearanceObserver = nil
       }
  }
```

Reply:
I had good luck with something that looked like this:
```
func startObserving() {
	kvoToken = NSApplication.shared.observe(\.effectiveAppearance) { (application, change) in
		guard self.bridge != nil else {
			return
		}
		// Do work
	}
}

func stopObserving() {
	kvoToken?.invalidate()

}
```